clang/test/CIR/CodeGen/const-array.c

```diff
@@ -0,0 +1,10 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -emit-cir %s -o - | FileCheck %s
+
+void foo() {
+  int a[10] = {1};
+}
+
+// CHECK: cir.func {{.*@foo}}
+// CHECK:   %0 = cir.alloca !cir.array<!s32i x 10>, cir.ptr <!cir.array<!s32i x 10>>, ["a"] {alignment = 16 : i64}
+// CHECK:   %1 = cir.const(#cir.const_array<[#cir.int<1> : !s32i], 9 zeros> : !cir.array<!s32i x 10>) : !cir.array<!s32i x 10>
```

Comment:
Thanks for working on this. I like how this simplifies and allows postponing all different initialization heuristics for structs (as you linked for unimplemented features).

Is there any special reason to print "9 zeros" instead of just "trailingzeros"? Are there other uses for the actual value?

I can see this being just implemented as a `UnitAttr`, which could optionally print "trailingzeros". Sounds like `mlir::cir::ArrayType` and `elts` can be used to compute how many trailing zeros there are when lowering?

Reply:
@bcardosolopes  
yes, it's better!  Though I used just `bool` instead of `UnitAttr`, hope it's ok)
So let's take a look again!